From 41d043cd0761c530fa6396d1981f355c01bfdc12 Mon Sep 17 00:00:00 2001
From: Brian Reading <brian.reading@gmail.com>
Date: Sat, 28 Jan 2023 20:11:17 -0600
Subject: [PATCH] podcasts_data: Use reqwest's default redirect policy

This should help with some podcasts that perform a lot of redirects.

The default policy uses 10 redirects for a tolerance, whereas we were
only using 5 with a custom policy before.
---
 podcasts-data/src/downloader.rs | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/podcasts-data/src/downloader.rs b/podcasts-data/src/downloader.rs
index 9949e24c..80a674e2 100644
--- a/podcasts-data/src/downloader.rs
+++ b/podcasts-data/src/downloader.rs
@@ -59,23 +59,7 @@ fn download_into(
     progress: Option<Arc<Mutex<dyn DownloadProgress>>>,
 ) -> Result<String, DownloadError> {
     info!("GET request to: {}", url);
-    // Haven't included the loop check as
-    // Steal the Stars would trigger it as
-    // it has a loop back before giving correct url
-    let policy = Policy::custom(|attempt| {
-        info!("Redirect Attempt URL: {:?}", attempt.url());
-        if attempt.previous().len() > 5 {
-            attempt.error("too many redirects")
-        } else if Some(attempt.url()) == attempt.previous().last() {
-            // avoid redirect loops
-            attempt.stop()
-        } else {
-            attempt.follow()
-        }
-    });
-
     let client = reqwest::blocking::Client::builder()
-        .redirect(policy)
         .referer(false)
         .user_agent(crate::USER_AGENT)
         .build()?;
-- 
GitLab

